src/Swashbuckle.AspNetCore.Filters/Examples/MvcOutputFormatter.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Swashbuckle.AspNetCore.Filters
+{
+    internal class MvcOutputFormatter
+    {
+        private readonly OutputFormatterSelector outputFormatterSelector;
+
+        public MvcOutputFormatter(IOptions<MvcOptions> options, ILoggerFactory loggerFactory)
```

Comment:
Hi again
There's quite a lot going on in this class (and in this constructor), and I'm currently looking at it in a web browser and not in Visual Studio, so I don't really understand what it's doing.

BUT - is this only needed for XML examples? If so, then I hope this constructor only gets called if the user is outputting XML examples. 99% of people use JSON examples and have no need for XML.

Reply:
We'll lazy instantiate it. It is indeed only for XML examples.

There's a lot of use of standard MS classes wrt OutputFormatter selection, taking from unit tests in dotnet/aspnetcore repo to construct it correctly and cover edge cases.

Reply:
It's lazy loaded now